anitya/lib/model.py

```diff
@@ -334,8 +364,42 @@ def validate_ecosystem_name(self, key, value):
     def versions(self):
         ''' Return list of all versions stored, sorted from newest to oldest.
         '''
-        return list(reversed(anitya.order_versions(
-            [v.version for v in self.versions_obj])))
+        version_class = self.get_version_class()
+        versions = [
+            version_class(version=v_obj.version, prefix=self.version_prefix)
+            for v_obj in self.versions_obj
+        ]
+        sorted_versions = reversed(sorted(versions))
+        return [v.version for v in sorted_versions]
+
+    def get_version_class(self):
+        """
+        Get the class for the version scheme used by this project.
+
+        This will take into account the defaults set in the ecosystem, backend,
+        and globally. The version scheme locations are checked in the following
+        order and the first non-null result is returned:
+
+        1. On the project itself in the ``version_scheme`` column.
+        2. The project's ecosystem default, if the project is part of one.
+        3. The project's backend default, if the backend defines one.
+        4. The global default defined in :data:`anitya.lib.versions.GLOBAL_DEFAULT`
+
+        Returns:
+            anitya.lib.versions.Version: A ``Version`` sub-class.
+        """
+        from .plugins import ECOSYSTEM_PLUGINS, BACKEND_PLUGINS, VERSION_PLUGINS
```

Comment:
Yup. I have a plan to fix them, but this PR actually makes it easier to do later (as it rips out one of the things defined in ``anitya/__init__.py``. I started to do it in a separate branch before this gets merged, but realized that having this in would make untangling the imports easier. This is very much a short-term work-around. A purge of in-function imports is coming.